core/src/main/scala/org/apache/spark/storage/BlockManager.scala

```diff
@@ -731,7 +731,14 @@ private[spark] class BlockManager(
       }
 
       if (data != null) {
-        return Some(ChunkedByteBuffer.fromManagedBuffer(data, chunkSize))
+        // SPARK-24307 undocumented "escape-hatch" in case there are any issues in converting to
+        // to ChunkedByteBuffer, to go back to old code-path.  Can be removed post Spark 2.4 if
+        // new path is stable.
+        if (conf.getBoolean("spark.fetchToNioBuffer", false)) {
```

Comment:
can we have a better prefix, rather than just spark. ?

Reply:
Since this condition is immutable, can we define a new variable whose value is assigned out of this method to reduce overhead?

Reply:
sure -- the fetch-to-disk conf is "spark.maxRemoteBlockSizeFetchToMem" which is why I stuck with just "spark." prefix.  Also on second thought, I will make the rest of it more specific too, as there is lots of "fetching" this doesn't effect.

how about "spark.network.remoteReadNioBufferConversion"?

Reply:
Maybe we'd better to rename that one "spark.maxRemoteBlockSizeFetchToMem" also ?